src/main/java/au/com/addstar/truehardcore/TrueHardcore.java

```diff
@@ -264,16 +267,25 @@ public void onEnable() {
         p = pm.getPlugin("ProtocolLib");
         if (p instanceof ProtocolLib) {
             ProtocolLibrary.getProtocolManager().addPacketListener(
-                  new PacketListener(
-                        this,
-                        ListenerPriority.NORMAL,
-                        PacketType.Play.Server.RESPAWN,
-                        PacketType.Play.Server.LOGIN));
+                    new PacketListener(
+                            this,
+                            ListenerPriority.NORMAL,
+                            PacketType.Play.Server.RESPAWN,
+                            PacketType.Play.Server.LOGIN));
             log("ProtocolLib found! Hardcore hearts will be enabled.");
         } else {
             log("ProtocolLib not found! Hardcore hearts will not work.");
         }
 
+        if (Bukkit.getPluginManager().getPlugin("PlaceholderAPI") != null) {
+            getLogger().info("PlaceholderAPI found! Registering placeholders...");
+            new PlaceholderAPI(this).register();
```

Comment:
Not a fan of this syntax, it's really an odd way to do this :thinking: not even sure this does what you expect and could be problematic.

Where did you get this syntax from? I don't see it anywhere in the docs and not sure why this is necessary. I don't think we need to register an expansion for PAPI, unless we wanted to reference TH values in other plugins?

To be able to resolve PAPI values in TH, we should be able to just call the `setPlaceholders()` method as described in the [PAPI docs](https://github.com/PlaceholderAPI/PlaceholderAPI/wiki/Hook-into-PlaceholderAPI#setting-placeholders-in-your-plugin).

Reply:
nevermind, I misunderstood this.. I thought `PlaceholderAPI` was referring to the PAPI plugin, not the TH class.